File: Grade_1/1-2/Q3.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main ()
{
    printf("202533713 김진서\n");

    int human, com; // 사람과 컴퓨터가 낸 가위, 바위, 보를 저장할 변수
    int diff; // human - com을 통해 승리자를 가려내기 위한 변수
    int win_human = 0, win_com = 0; // 승리 횟수를 저장할 변수

    srand(time(NULL));

    printf("가위바위보 게임\n");
    printf("컴퓨터 : 0승 0패, 당신 : 0승 0패\n");

    for (int cnt = 1; cnt <= 5; cnt++)
    {
        printf("(라운드 %d)\n", cnt);

        while (1)
        {
            com = rand(); // 랜덤한 0 ~ 32767까지의 값이 들어옴

            if (com == 0 || com == 1 || com == 2) // 0, 1, 2 값이 나왔을 때만 사용하기 위함
            {
                printf("컴퓨터가 결정했습니다.\n");

                printf("무엇을 내시겠습니까? (가위: 0, 바위: 1, 보: 2)");
                scanf("%d", &human);

                if (human == com)
                    printf("무승부입니다. 다시 시작합니다.\n");
                else
                    break;
            }
        }

        diff = human - com; // 결과값: 1, -1, 2, -2

        // 사람이 이겼을 때 (바위 - 가위, 보 - 바위: 1)
        if (diff == 1)
        {
            if (human == 2)
                printf("컴퓨터는 바위, 당신은 보, 당신이 이겼습니다.\n");
            else
                printf("컴퓨터는 가위, 당신은 바위, 당신이 이겼습니다.\n");
            win_human += 1;
        }

        // 컴퓨터가 이겼을 때 (바위 - 가위, 보 - 바위: -1)
        if (diff == -1)
        {
            if (com == 2)
                printf("컴퓨터는 보, 당신은 바위, 컴퓨터가 이겼습니다.\n");
            else
                printf("컴퓨터는 바위, 당신은 가위, 컴퓨터가 이겼습니다.\n");
            win_com += 1;
        }

        // 가위가 보를 이겼을 때 (2 또는 -2)
        if (diff == 2 || diff == -2)
        {
            if (human == 2)
            {
                printf("컴퓨터는 가위, 당신은 보, 컴퓨터가 이겼습니다.\n");
                win_com += 1;
            }

            if (com == 2)
            {
                printf("컴퓨터는 보, 당신은 가위, 당신이 이겼습니다.\n");
                win_human += 1;
            }
        }
        
        // 승 = win_humanm, win_com     패 = 진행된 라운드 - 승리 횟수
        printf("컴퓨터 : %d승 %d패, 당신 : %d승 %d패\n\n", win_com, (cnt - win_com), win_human, (cnt - win_human));
            
    }
}

File: Grade_1/1-2/homework2/Q3.py
import random

## 각 숫자에 맞는 가위, 바위, 보를 key: value 로 매칭
SRP = {0: "가위", 1: "바위", 2: "보"}

win_com, win_human = 0, 0

print("202533713 김진서")

print("가위바위보 게임")
print("컴퓨터 : 0승 0패, 당신 : 0승 0패")

for cnt in range(1, 6):
    print(f"(라운드 {cnt})")

    while True:
        com = (random.randint(0, 2)) ## 0 ~ 2까지의 값만 받음
        print("컴퓨터가 결정했습니다.")

        human = int(input("무엇을 내시겠습니까? (가위 : 0, 바위 : 1, 보 : 2)"))

        if com == human:
            print("비겼습니다. 다시 시작합니다.")
            continue

        break
    
    ### SRP(dict)을 통해 입력받은 human, com 변수의 정수값으로 가위, 바위, 보의 문자값을 매칭
    ## 사람이 이겼을 경우 (보 - 바위, 바위 - 가위)
    if (human - com == 1):
        print(f"컴퓨터는 {SRP.get(com)}, 당신은 {SRP.get(human)}, 당신이 이겼습니다.")
        win_human += 1

    ## 컴퓨터 이겼을 경우 (보 - 바위, 바위 - 가위)
    if (com - human == 1):
        print(f"컴퓨터는 {SRP.get(com)}, 당신은 {SRP.get(human)}, 컴퓨터가 이겼습니다.")
        win_com += 1

    ## 사람이 이겼을 경우 (가위 - 보)
    if (com - human == 2):
        print(f"컴퓨터는 {SRP.get(com)}, 당신은 {SRP.get(human)}, 당신이 이겼습니다.")
        win_human += 1

    ## 컴퓨터 이겼을 경우 (가위 - 보)
    if (human - com == 2):
        print(f"컴퓨터는 {SRP.get(com)}, 당신은 {SRP.get(human)}, 컴퓨터가 이겼습니다.")
        win_com += 1
    
    print(f"컴퓨터 : {win_com}승 {cnt - win_com}패, 당신 : {win_human}승 {cnt - win_human}패\n")
    